google/cloud/sql/connector/rate_limiter.py

```diff
@@ -0,0 +1,95 @@
+"""
+Copyright 2021 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import asyncio
+import math
+import time
+
+
+class AsyncRateLimiter(object):
+    """
+    An asyncio-compatible rate limiter which uses the Token Bucket algorithm
+    (https://en.wikipedia.org/wiki/Token_bucket) to limit the number of function calls over a time interval using an event queue.
+
+    :type burst_size: int
+    :param: burst_size:
+        the maximum capacity of the bucket will store at any one time.
+        Default: 1
+
+    :type interval: int
+    :param: interval:
+        The period of time over which a number of calls equal to burst_size are allowed to complete. Default: 60s
+
+    :type loop: asyncio.AbstractEventLoop
+    :param: loop:
+        The event loop to use. If not provided, the default event loop will be used.
+
+
+    """
+
+    def __init__(
+        self,
+        burst_size: int = 1,
```

Comment:
Do we ever need burst_size to be greater than one? It think we should never need to acquire more than one token at a time